tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetURITest.scala

```diff
@@ -14,19 +15,20 @@ class PublicAssetURITest {
   def fromShouldReturnCorrectPublicURI(): Unit = {
     val prefixUri = new URI("http://localhost:8080")
     val assetId = PublicAssetId(UUID.fromString("a0f7b4b3-0b1b-4b3b-8b3b-0b1b4b3b8b3b"))
-    val account = AccountId("aHR0cHM6Ly93d3cuYmFzZTY0ZW5jb2RlLm9yZy8")
-    val result: PublicURI = PublicURI.from(assetId, account, prefixUri)
+    val username = Username.of("username@localhost")
+    val result: PublicURI = PublicURI.from(assetId, username, prefixUri)
 
-    assertThat(result.value.toString).isEqualTo("http://localhost:8080/publicAsset/aHR0cHM6Ly93d3cuYmFzZTY0ZW5jb2RlLm9yZy8/a0f7b4b3-0b1b-4b3b-8b3b-0b1b4b3b8b3b")
+    assertThat(result.value.toString).isEqualTo("http://localhost:8080/publicAsset/username@localhost/a0f7b4b3-0b1b-4b3b-8b3b-0b1b4b3b8b3b")
   }
 
   @Test
   def fromShouldReturnCorrectPublicURIWhenPrefixUriContainSeveralPathSegments(): Unit = {
     val prefixUri = new URI("http://localhost:8080/jmap/abc")
     val assetId = PublicAssetId(UUID.fromString("a0f7b4b3-0b1b-4b3b-8b3b-0b1b4b3b8b3b"))
     val account = AccountId("aHR0cHM6Ly93d3cuYmFzZTY0ZW5jb2RlLm9yZy8")
-    val result: PublicURI = PublicURI.from(assetId, account, prefixUri)
+    val username = Username.of("username@localhost")
+    val result: PublicURI = PublicURI.from(assetId, username, prefixUri)
 
-    assertThat(result.value.toString).isEqualTo("http://localhost:8080/jmap/abc/publicAsset/aHR0cHM6Ly93d3cuYmFzZTY0ZW5jb2RlLm9yZy8/a0f7b4b3-0b1b-4b3b-8b3b-0b1b4b3b8b3b")
+    assertThat(result.value.toString).isEqualTo("http://localhost:8080/jmap/abc/publicAsset/username@localhost/a0f7b4b3-0b1b-4b3b-8b3b-0b1b4b3b8b3b")
```

Comment:
Shouldn't we url encode the "@" ?

Reply:
No need
https://stackoverflow.com/questions/4669692/valid-characters-for-directory-part-of-a-url-for-short-links

Reply:
We do not encode/decode the username in the webadmin route for example. So I think it is fine to not encode the username here too?